### PR TITLE
Discover kubearmor policies on per process

### PR DESCRIPTION
**Pull request by seswarrajan**
Signed-off-by: Eswar Rajan Subramanian <eswar@accuknox.com>

### CODE REVIEW

File: src/systempolicy/helperFunctions.go [hunk 1]
```diff
@@ -152,3 +152,25 @@ func FilterSystemLogsByConfig(logs []types.KnoxSystemLog, pods []types.Pod) []ty
...
 }
+
+func GetWPFSSources() []string {
+	res, _, err := libs.GetWorkloadProcessFileSet(CfgDB, types.WorkloadProcessFileSet{})
+	if err != nil {
+		log.Error().Msgf("cudnot fetch WPFS err=%s", err.Error())
+		return nil
+	}
+
+	if res != nil {
```
Comment (nyrahul) [inline src/systempolicy/helperFunctions.go]:
```suggestion
	if res == nil {
	    return nil
	}
```
This helps in reducing nesting.

File: src/systempolicy/systemPolicy.go [hunk 2]
```diff
@@ -286,29 +287,83 @@ func WriteSystemPoliciesToFile(namespace, clustername, labels string) {
...
 		response.Kubearmorpolicy = append(response.Kubearmorpolicy, &kubearmorpolicy)
 	}
+
+	// system policy for VM
+	for i := range kubearmorVMPolicies {
+		kubearmorpolicy := wpb.KubeArmorPolicy{}
+
+		val, err := json.Marshal(&kubearmorVMPolicies[i])
+		if err != nil {
+			log.Error().Msg(err.Error())
```
Comment (nyrahul) [inline src/systempolicy/systemPolicy.go]: It is better to `continue` in this failure case. Also please add a text label before the error message. 
```suggestion
			log.Error().Msg("kubearmorVMPolicy json marshal failed err=%v", err.Error())
			continue
```

**Review by nyrahul — CHANGES_REQUESTED** [review]
Thanks for the PR. PFA my review comments.

**Review by nyrahul — APPROVED** [review]
LGTM

File: src/systempolicy/helperFunctions.go [hunk 3]
```diff
@@ -152,3 +152,25 @@ func FilterSystemLogsByConfig(logs []types.KnoxSystemLog, pods []types.Pod) []ty
 
 	return filteredLogs
 }
+
+func GetWPFSSources() []string {
+	res, _, err := libs.GetWorkloadProcessFileSet(CfgDB, types.WorkloadProcessFileSet{})
+	if err != nil {
+		log.Error().Msgf("cudnot fetch WPFS err=%s", err.Error())
```
Comment (Ankurk99) [inline src/systempolicy/helperFunctions.go:159]: minor nit

**Pull request merged**